src/Artsy/Analytics/trackingMiddleware.ts

```diff
@@ -0,0 +1,77 @@
+import { trackExperimentViewed } from "Artsy/Analytics/trackExperimentViewed"
+import { getENV } from "Utils/getENV"
+import createLogger from "Utils/logger"
+
+const logger = createLogger("Artsy/Analytics/trackingMiddleware")
+
+const ActionTypes = {
+  UPDATE_LOCATION: "@@farce/UPDATE_LOCATION",
+
+  /**
+   * Unused Farce/Found router events.
+   * TODO: Remove once we're sure we don't need to intercept anything else or
+   * after client side routing launches.
+   */
+  // INIT: "@@farce/INIT",
+  // PUSH: "@@farce/PUSH",
+  // REPLACE: "@@farce/REPLACE",
+  // TRANSITION: "@@farce/TRANSITION",
+  // GO: "@@farce/GO",
+  // CREATE_HREF: "@@farce/CREATE_HREF",
+  // CREATE_LOCATION: "@@farce/CREATE_LOCATION",
+  // DISPOSE: "@@farce/DISPOSE",
+}
+
+/**
+ * PageView tracking middleware for use in our router apps.
+ *
+ * @see Artsy/Router/buildClientApp.tsx for mount point
+ */
+export function trackingMiddleware() {
+  return _store => next => action => {
+    const { type, payload } = action
+
+    switch (type) {
+      case ActionTypes.UPDATE_LOCATION: {
```

Comment:
Any app that is built using our routing framework will now have automatic page-view tracking on URL changes.

Reply:
This whole file seems 👍 to me.